### PR TITLE
p5.ts 1.8

### PR DESCRIPTION
**Pull request by asukaminato0721**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/p5/src/webgl/3d_primitives.d.ts [hunk 1]
```diff
@@ -4,6 +4,60 @@ import * as p5 from '../../index';
...
+         *   Clears the resources of a model to free up browser
+         *   memory. A model whose resources have been cleared
+         *   can still be drawn, but the first time it is drawn
+         *   again, it might take longer. This method works on
+         *   models generated with buildGeometry() as well as
+         *   those loaded from loadModel().
+         *   @param The geometry whose resources should be
+         *   freed
+         */
+        freeGeometry(The: Geometry): void;
```
Comment (lindapaiste) [inline types/p5/src/webgl/3d_primitives.d.ts:59]: variable name `The` is a mistake, and was fixed by https://github.com/processing/p5.js/commit/6be446be07aa073922708f7f80636f83094bd373

File: types/p5/src/typography/loading_displaying.d.ts [hunk 2]
```diff
@@ -5,92 +5,119 @@ import * as p5 from '../../index';
...
+         *   p5.Font object after it loads.
+         *   @param [failureCallback] function called with the
+         *   error Event object if the font fails to load.
+         *   @return p5.Font object.
          */
-        loadFont(path: string, callback?: (...args: any[]) => any, onError?: (...args: any[]) => any): Font;
+        loadFont(
+            path: string,
+            successCallback?: (...args: any[]) => any,
+            failureCallback?: (...args: any[]) => any
```
Comment (lindapaiste) [inline types/p5/src/typography/loading_displaying.d.ts:44]: I'd love to see some actual types here.

File: types/p5/src/color/setting.d.ts [hunk 3]
```diff
@@ -4,6 +4,64 @@ import * as p5 from '../../index';
...
+         *
+         *   - invert: A boolean specifying whether or not to
+         *   mask the areas not filled by the mask shape.
+         *   Defaults to false.
+         *   @param callback A function that draws the mask
+         *   shape.
+         *   @param [options] An object containing clip
+         *   settings.
+         */
+        clip(callback: (...args: any[]) => any, options?: object): void;
```
Comment (lindapaiste) [inline types/p5/src/color/setting.d.ts:63]: What are the properties of the `options` object?  Does the `callback` return a value?  What are the arguments of the `callback`?

File: types/p5/src/color/setting.d.ts [hunk 4]
```diff
@@ -4,6 +4,64 @@ import * as p5 from '../../index';
...
+         *
+         *   Options can include:
+         *
+         *   - invert: A boolean specifying whether or not to
+         *   mask the areas not filled by the mask shape.
+         *   Defaults to false.
+         *   @param [options] An object containing clip
+         *   settings.
+         */
+        beginClip(options?: object): void;
```
Comment (lindapaiste) [inline types/p5/src/color/setting.d.ts:29]: What are the allowed `options`?  I know that this is an auto-generated types file and that change needs to be made in the root p5.js package but I would love to see a `@typedef` of `ClipOptions`!

File: types/p5/src/events/keyboard.d.ts [hunk 5]
```diff
@@ -35,7 +35,7 @@ declare module '../../index' {
          *   @param [event] optional KeyboardEvent callback
          *   argument.
          */
-        keyPressed(event?: object): void;
+        keyPressed(event?: KeyboardEvent): void;
```
Comment (lindapaiste) [inline types/p5/src/events/keyboard.d.ts:38]: ⭐

These sorts of changes are hugely important!

File: types/p5/src/image/loading_displaying.d.ts [hunk 6]
```diff
@@ -5,144 +5,174 @@ import * as p5 from '../../index';
...
-         *   notificationID, a string that specifies the
-         *   notification DOM element id. By default it’s
+         *   notification will live. By default it's 0, meaning
+         *   the notification will never be removed;
+         *   notificationID, a String that specifies the id of
+         *   the notification's DOM element. By default it’s
          *   'progressBar’.
          */
-        saveGif(filename: string, duration: number, options: object): void;
+        saveGif(filename: string, duration: number, options?: object): void;
```
Comment (lindapaiste) [inline types/p5/src/image/loading_displaying.d.ts:73]: This is another object where I would love to see us define the `options` interface.

**Review by lindapaiste — COMMENTED** [review]
Some quick thoughts:

- The "Huge Change" flag is largely due to changes to the JSDoc text, which is not important to review as it's copied from the base package.
- Changes to the types such as new functions and optional arguments should ideally have corresponding additions to the tests, but the existing tests here are scant.
- There are a some very positive changes which move from `object` to specific DOM objects, like `HTMLCanvasElement`, `MouseEvent`, etc.

**Comment by asukaminato0721** [comment]
https://github.com/processing/p5.js/blob/7e2610e032b0d646920c671122f7f03b6169f66f/rfc_p5js_2.md since there will be a 2 version, a lot of problem will be solved.

**Pull request closed without merging**